File: drestaurant-apps/drestaurant-monolith/src/main/java/com/drestaurant/query/handler/CustomerEventHandler.java
package com.drestaurant.query.handler;

import org.axonframework.config.ProcessingGroup;
import org.axonframework.eventhandling.EventHandler;
import org.axonframework.eventsourcing.SequenceNumber;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

import com.drestaurant.customer.domain.api.CustomerCreatedEvent;
import com.drestaurant.query.model.CustomerEntity;
import com.drestaurant.query.repository.CustomerRepository;

@ProcessingGroup("default")
@Component
class CustomerEventHandler {

	private CustomerRepository repository;

	@Autowired
	public CustomerEventHandler(CustomerRepository repository) {
		this.repository = repository;
	}

	@EventHandler
	public void handle(CustomerCreatedEvent event, @SequenceNumber Long aggregateVersion) {
		repository.save(new CustomerEntity(event.getAggregateIdentifier(), aggregateVersion, event.getName().getFirstName(), event.getName().getLastName(), event.getOrderLimit().getAmount()));
	}

}

File: drestaurant-apps/drestaurant-monolith/src/main/kotlin/com/drestaurant/query/handler/CustomerEventHandler.kt
package com.drestaurant.query.handler

import com.drestaurant.customer.domain.api.CustomerCreatedEvent
import com.drestaurant.query.model.CustomerEntity
import com.drestaurant.query.repository.CustomerRepository
import org.axonframework.config.ProcessingGroup
import org.axonframework.eventhandling.EventHandler
import org.axonframework.eventsourcing.SequenceNumber
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.messaging.simp.SimpMessageSendingOperations
import org.springframework.stereotype.Component

@ProcessingGroup("default")
@Component
internal class CustomerEventHandler @Autowired constructor(private val repository: CustomerRepository, private val messagingTemplate: SimpMessageSendingOperations) {

    @EventHandler
    fun handle(event: CustomerCreatedEvent, @SequenceNumber aggregateVersion: Long) {
        repository.save(CustomerEntity(event.aggregateIdentifier, aggregateVersion, event.name.firstName, event.name.lastName, event.orderLimit.amount));
        messagingTemplate.convertAndSend("/topic/customers.updates", event);
    }

}
